.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: CI-test
+
+# Controls when the action will run.
+on: [push, pull_request]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v4
+
+      # setup python with different versions
+      - name: setup python version ${{ matrix.python-version}}
```

Comment:
why do you need different versions of python

Reply:
I read up it is a good idea to test for multiple os. since it is a small project I will review it